ThreeBotPackages/tfgrid/threebot_provisioning/chatflows/threebot_reservation.py

```diff
@@ -0,0 +1,135 @@
+    env = dict()
+    secret_env = dict()
+
+    if not name or not email:
+        bot.md_show("Username or email not found in session. Please log in properly")
+
+    user_choice = bot.single_choice("create a new 3bot or restore existing 3bot?", choose)
+    identity = explorer.actors_all.phonebook.get(name=name, email=email)
+    identity_pubkey = identity.pubkey
+    if user_choice == "Restore":
+        password = bot.secret_ask("please add restore password?")
+        hash_restore = nacl.hash.blake2b(password.encode(), key=identity_pubkey.encode()).decode()
+
+    # ask user about corex user:password and ssh-key to give him full access to his container
+    pub_key = bot.string_ask(
+        "Please add your public ssh-key (that will allow you to access the deployed container using ssh) "
+    )
+    user_corex = bot.string_ask(
+        "username of your coreX (that will allow you to be secure when access from web browser)"
+    )
+    password = bot.secret_ask("password of your coreX (that will allow you to be secure when access from web browser)")
+
+    # create new reservation
+    reservation = j.sal.zosv2.reservation_create()
+
+    ip_version = bot.single_choice("choose your IP version", ips)
+    node_selected = j.sal.chatflow.nodes_get(1)[0]
+
+    # Encrypt AWS ID and AWS Secret to send it in secret env
+    aws_id_encrypt = j.sal.zosv2.container.encrypt_secret(node_selected.node_id, AWS_ID)
+    aws_secret_encrypt = j.sal.zosv2.container.encrypt_secret(node_selected.node_id, AWS_SECRET)
+
+    # Create network of reservation and add peers
+    if user_choice == "Restore":
+        hash_encrypt = j.sal.zosv2.container.encrypt_secret(node_selected.node_id, hash_restore)
+        env.update({"restore": "True"})
+        secret_env.update({"HASH": hash_encrypt})
+
+    reservation, config = j.sal.chatflow.network_configure(
+        bot, reservation, [node_selected], customer_tid=identity.id, ip_version=ip_version
+    )
+
+    ip_address = config["ip_addresses"][0]
+
+    backup = bot.single_choice("Auto-backup 3bot?", ["Yes", "No"])
+    if backup == "Yes":
+        password = bot.secret_ask("Backup password? Don't forget to save this password to be able to restore")
+        hash_backup = nacl.hash.blake2b(password.encode(), key=identity_pubkey.encode()).decode()
+        hash_encrypt = j.sal.zosv2.container.encrypt_secret(node_selected.node_id, hash_backup)
+        secret_env.update({"HASH": hash_encrypt})
+        env.update({"backup": "True", "FOLDER": folder_name})
+
+    env.update({"corex_password": password, "corex_user": user_corex, "pub_key": pub_key})
+    secret_env.update({"AWS_ID": aws_id_encrypt, "AWS_SECRET": aws_secret_encrypt})
+
+    container_flist = "https://hub.grid.tf/bola_nasr_1/threefoldtech-3bot-corex.flist"
+    entry_point = "/usr/bin/zinit init -d"
+    storage_url = "zdb://hub.grid.tf:9900"
+
+    # Add volume and create container schema
+    expiration = j.data.time.epoch + (60 * 60 * 24)
```

Comment:
we can either ask or not, but let's make the expiration only set once, not here and line: 107